FrontEnd/src/HomePage/RecommendationDB.js

```diff
@@ -40,87 +71,255 @@ const addToFavoriteDocs = async(userUID, favorited, post) =>{
       likedAt: new Date(),
     });
   }
-}
+};
 
-const isLiked = (userUID, postID, callback) =>{
-  const favoritesRef = collection(database, DATABASE_FOLDER_NAME, userUID, FAVORITES_COLLECTION)
+const isLiked = (userUID, postID, callback) => {
+  const favoritesRef = collection(
+    database,
+    DATABASE_FOLDER_NAME,
+    userUID,
+    FAVORITES_COLLECTION
+  );
   const q = query(favoritesRef, where("post.postID", "==", postID));
   const unsubscribe = onSnapshot(q, (querySnapshot) => {
     callback(!querySnapshot.empty);
-});
+  });
 
-return unsubscribe
-}
+  return unsubscribe;
+};
 
-const postsFromFavorites = async(userUID) =>{
-  try{
-    const favoritesRef = collection(database, DATABASE_FOLDER_NAME, userUID, FAVORITES_COLLECTION)
-    const favoritePosts = await getDocs(favoritesRef)
-    const favoriteVendorsID = []
+//vendors users have interacted with in some way (by likes or appointments)
+const getRecommendedVendors = async (userID, vendorID) => {
```

Comment:
Great job on implementing recommendation algorithm!! Sharing some thoughts --
Normally we want to let js side to only handle user-interaction. For computing-heavy job, it's better to move the entire logic to backend. The recommended API format could be: JS: getRecommendedPostID(userUID) -> Array(recommendedPost).
The execution efficiency for JS is normally way lower than backend languages (c++/Python), also it's easier to get efficiency optimized on backend. 
Another reason for the frontend-backend design is that we can take the calculation steps offline. So when user come to make a request, the recommended posts could be pre-calculated.
I know moving this logic totally to backend would add a lot of complexity, so just sharing my thoughts here. You don't need to apply it right now. :)